online-shop/src/app/page-product-details/page-product-details.component.ts

```diff
@@ -0,0 +1,57 @@
+import {Component, Input, OnInit} from '@angular/core';
+import {Product} from '../models/product';
+import {ActivatedRoute, Router} from '@angular/router';
+import {ProductService} from '../services/product.service';
+import {ShoppingCartService} from '../services/shopping-cart.service';
+import {AuthService} from '../services/auth.service';
+import {MatDialog} from '@angular/material/dialog';
+import {ConfirmationModalComponent} from './confirmation-modal/confirmation-modal.component';
+
+@Component({
+  selector: 'app-page-product-details',
+  templateUrl: './page-product-details.component.html',
+  styleUrls: ['./page-product-details.component.css']
+})
+export class PageProductDetailsComponent implements OnInit {
+
+  @Input() product: Product;
+  wasAddedToTheCart: boolean;
+
+  constructor(
+    public authService: AuthService,
+    public dialog: MatDialog,
+    private route: ActivatedRoute,
+    private router: Router,
+    private productService: ProductService,
+    private shoppingCartService: ShoppingCartService) {
+  }
+
+  ngOnInit() {
+    this.wasAddedToTheCart = false;
+    this.getProduct();
+  }
+
+  getProduct(): void {
+    const id = parseInt(this.route.snapshot.paramMap.get('id'), 10);
+    this.productService.getProduct(id).subscribe(product => this.product = product);
+  }
+
+  delete(): void {
+    const dialogRef = this.dialog.open(ConfirmationModalComponent, {
+      width: '500px',
```

Comment:
Small hint: in general it is good to avoid hardcoding sizes. Percentage based sizes or responsive sizes are generally better... In this case though, it is fine like this.